{
  cleanEntity(persistencePackage.getEntity());
  if (persistencePackage.getEntity().isValidationFailure()) {
    return new PersistenceResponse().withEntity(persistencePackage.getEntity());
  }
  try {
    PersistenceManager persistenceManager=(PersistenceManager)applicationContext.getBean(persistenceManagerRef);
    persistenceManager.setTargetMode(TargetModeType.SANDBOX);
    return persistenceManager.add(persistencePackage);
  }
 catch (  ServiceException e) {
    if (e instanceof ValidationException) {
      LOG.warn("Not saving entity as it has failed validation");
      return new PersistenceResponse().withEntity(((ValidationException)e).getEntity());
    }
 else     if (e.getCause() instanceof ValidationException) {
      LOG.warn("Not saving entity as it has failed validation");
      return new PersistenceResponse().withEntity(((ValidationException)e.getCause()).getEntity());
    }
    String message=exploitProtectionService.cleanString(e.getMessage());
    LOG.error("Problem adding new " + persistencePackage.getCeilingEntityFullyQualifiedClassname(),e);
    throw recreateSpecificServiceException(e,message,e.getCause());
  }
}
